![alt text](https://secure.meetupstatic.com/photos/event/c/9/7/a/600_487911578.jpeg)

# Assignment 3

You joined Daldi GmbH as a junior data scientist. Daldi GmbH is a supermarket chain in Austria. You need to prove that hiring a data scientist in their company was a good idea. Your manager has provided you with the following dataset (daldi_sales.csv).
You will have to present your results to the CEO. You are free to explore the data, as you wish. If you need a starting point, your Manager recommends answering the following question:

What is the most profitable Chain/City?

How effective is our Membership Program?

Which products are most succesful among women/men?


**Description Data**


invoice_id: Computer generated sales slip invoice identification number

branch: Branch of supercenter (3 branches are available identified by A, B and C).

city: Location of supercenters

customer_type: Type of customers, recorded by Members for customers using member card and Normal for without member card.

gender: Gender type of customer

product_type: General item categorization groups - Electronic accessories, Fashion accessories, Food and beverages, Health and beauty, Home and lifestyle, Sports and travel

unit_price: Price of each product in €

quantity: Number of products purchased by customer


total: Total price including tax

Payment: Payment used by customer for purchase (3 methods are available – Cash, Credit card and Ewallet)

Profit: profit in €

customer_rating: Customer stratification rating on their overall shopping experience (On a scale of 1 to 10)

**Submission**

For submission, either prepare a 5 min Powerpoint or a .ipynb (Google Collab) file with a description of the things done.

**To download the Google Collab, Press:**


File > Download .ipynb

**Upload Here your Solutions**
[https://forms.gle/Ngf9W6ebcB2x2wNF8](https://forms.gle/Ngf9W6ebcB2x2wNF8)


# Download and Load Data

In [0]:
import pandas as pd
import numpy as np
import plotly.express as px

In [0]:
!wget http://mlgraz.at/course_files/week3/daldi_sales.csv

--2020-04-16 11:43:16--  http://mlgraz.at/course_files/week3/daldi_sales.csv
Resolving mlgraz.at (mlgraz.at)... 81.19.159.92
Connecting to mlgraz.at (mlgraz.at)|81.19.159.92|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: http://www.mlgraz.at/course_files/week3/daldi_sales.csv [following]
--2020-04-16 11:43:17--  http://www.mlgraz.at/course_files/week3/daldi_sales.csv
Resolving www.mlgraz.at (www.mlgraz.at)... 81.19.159.92
Connecting to www.mlgraz.at (www.mlgraz.at)|81.19.159.92|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://www.mlgraz.at/course_files/week3/daldi_sales.csv [following]
--2020-04-16 11:43:17--  https://www.mlgraz.at/course_files/week3/daldi_sales.csv
Connecting to www.mlgraz.at (www.mlgraz.at)|81.19.159.92|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 86054 (84K) [text/csv]
Saving to: ‘daldi_sales.csv.1’

daldi_sales.csv.1   100%[===================>]  84.04K   27

In [0]:
df = pd.read_csv("daldi_sales.csv")

# The most profitable city.







In [0]:
branches_profit = df.groupby("city").sum().reset_index()
branches_profit.sort_values(by=["profit"],ascending=False, inplace=True)

fig_profit = px.bar(data_frame=branches_profit, x="city", y="profit", orientation="v",  title="Profit overview in all branches")
fig_profit.show()

#The effectivness of the membership programm.






In [0]:
membership_profit_sum = df.groupby("customer_type").sum().reset_index()

fig_membership_profit = px.bar(data_frame=membership_profit_sum, x="customer_type", y="profit", orientation="v",  title="Profit overview for the customers with/without the memebrship.")
fig_membership_profit.show()

#Which products are most succesful among women/men?



In [0]:
gender_profit = df.groupby(["gender", "product_type"]).quantity.sum().reset_index()
female_max_quantity = gender_profit[gender_profit.gender == "Female"].quantity.max()
male_max_quantity = gender_profit[gender_profit.gender == "Male"].quantity.max()

gender_profit[gender_profit.quantity.isin([female_max_quantity, male_max_quantity])]

gender         product_type  quantity
1  Female  Fashion accessories       530
9    Male    Health and beauty       511



*   The most succesful product among women is *Fashion accessories* (bought 530 times).
*   The most succesful product among men is Health and beauty (bought 511 times.



#Which type of payment is most popular?




In [0]:
payment_count = df.payment.value_counts().reset_index().rename(columns={"index": "payment type", "payment": "frequency"})
px.bar(data_frame=payment_count, x="payment type" , y="frequency", title="Frequency of using particular payment type.")

*Ewallet* is the most frequently used payment type. But does it bring the most income?

In [0]:
payment = df.groupby("payment").total.sum().reset_index()
payment.sort_values(by=['total'],ascending=False, inplace=True)
px.bar(data_frame=payment, x="payment", y="total", orientation="v", title="Total amount of the income distributed over the payment types.")

Although, *Ewallet* is the most often used payment type, *Cash* still brings the most income.

#How many taxes did we pay?




In [0]:
total_without_taxes = df.unit_price*df.quantity
df["total_without_taxes"] = total_without_taxes

taxes_paid = df.total.sum()-df.total_without_taxes.sum()
income = df.total.sum()
percentage = income / taxes_paid

data = {'income':  [ income], "taxes_paid": [taxes_paid], "percentage": [percentage] }
df_income = pd.DataFrame (data, columns = ['income', 'taxes_paid', 'percentage'])
df_income

income  taxes_paid  percentage
0  322966.82    15379.44   20.999908

We paid **15379.44** EUR taxes which is about 21% of the income of **322966.82** EUR .